airbyte-cdk/python/airbyte_cdk/sources/streams/http/http_client.py

```diff
@@ -0,0 +1,257 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import logging
+import os
+import urllib
+from pathlib import Path
+from typing import Any, Mapping, Optional, Tuple, Union
+
+import requests
+import requests_cache
+from airbyte_cdk.models import FailureType
+from airbyte_cdk.sources.http_config import MAX_CONNECTION_POOL_SIZE
+from airbyte_cdk.sources.streams.call_rate import APIBudget, CachedLimiterSession, LimiterSession
+from airbyte_cdk.utils.constants import ENV_REQUEST_CACHE_PATH
+from airbyte_cdk.utils.traced_exception import AirbyteTracedException
+from requests.auth import AuthBase
+
+from .decoders import Decoder, JsonDecoder
+from .error_handlers import DefaultBackoffStrategy, HttpStatusErrorHandler, JsonErrorMessageParser, ResponseAction
+from .exceptions import DefaultBackoffException, RequestBodyException, UserDefinedBackoffException
+from .rate_limiting import default_backoff_handler, user_defined_backoff_handler
+
+BODY_REQUEST_METHODS = ("GET", "POST", "PUT", "PATCH")
+
+
+class HttpClient:
+    def __init__(
+        self,
+        stream_name: str,
+        logger: logging.Logger,
+        api_budget: Optional[APIBudget] = None,
+        session: Optional[Union[requests.Session, requests_cache.CachedSession]] = None,
+        authenticator: Optional[AuthBase] = None,
```

Comment:
Just a note that this adds another thing that is "requests lib specific"

Reply:
connectors should move off the deprecated authenticators. Good callout on needing to abstract away the dependency on the requests library though. [Added a note in the CDK revamp doc](https://docs.google.com/document/d/1i0CypW8yTmCR_-poez8odKbFAZd4EDwHljQGlak8XXo/edit#heading=h.7fwsbtmt7ptw)

Reply:
Noted. Not scoped as part of this PR so will revisit down the line.